### ISSUE TITLE
Tkinter plugin erroring. Works fine on regular Nuitka.

### ISSUE THREAD

**Issue by Jaxx7594** [Nuitka/Nuitka#2730]
I don't get any errors when running the exact same command on my computer, it only errors when using Nuitka Options to compile. Heres my action yml: https://github.com/Jaxx7594/test-python-script-to-exe/blob/main/.github/workflows/test.yml. I also tested enable-plugin***s*** as shown in the examples instead of enable-plugin, and it simply does nothing. The plugin isn't enabled, and the tcl files aren't included.
```
Nuitka-Options: Used command line options: --standalone --disable-console --enable-plugin=tk-inter --windows-icon-from-ico=images/icon.png --remove-output --mingw64 --assume-yes-for-downloads --output-dir=build --script-name=main.py
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.8\x64\Lib\site-packages\nuitka\plugins\Plugins.py", line 82, in withPluginProblemReporting
    yield
  File "C:\hostedtoolcache\windows\Python\3.11.8\x64\Lib\site-packages\nuitka\plugins\Plugins.py", line 123, in _addActivePlugin
    plugin_instance = plugin_class(**plugin_args)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: NuitkaPluginTkinter.__init__() missing 2 required positional arguments: 'tcl_library_dir' and 'tk_library_dir'
FATAL: tk-inter: Plugin issue while working on 'Plugin initialization failed'. Please report the bug with the above traceback included.
Error: Process completed with exit code 1.
```

**Comment by kayhayen** [comment]
The code that does the parsing now lives in Nuitka itself. Therefore only factory will work now with your config, and main once I push this out as a hotfix, very likely tomorrow. I am closing this early, as it seems not generally relevant. Actions do not use a short form of the argument in any example. In Nuitka we prefer the --enable-plugin abbreviation, because you can give arguments multiple times, no so in yaml, there the "," separation for this option and the newline separation for others makes it necessary to specify it in plural. For nuitka help the plural is the default simply because we can do it like that.

**Comment by Jaxx7594** [comment]
It seems its even more broken now. There isn't an error, but its the same issue I mentioned previously about --enable-plugins (doesn't do anything), except it happens for both --enable-plugin and --enable-plugins now. This is an image of the result of a build using --enable-plugins, as you can see, there is no tcl folders or files. strangely enough, the main executable is also about 15% smaller than building on my computer using the same command, compiler, everything. Not sure why that is, just seemed strange to me.
<img width="839" alt="image" src="https://github.com/Nuitka/Nuitka/assets/101913901/ba40d840-a280-4348-a7f1-43d07d396b08">

**Comment by kayhayen** [comment]
I am not sure what you think that screenshot is going to help with. I indeed removed a difference between those two, I wasn't sure what actually happens, but I corrected the crash I saw, under the theory that abbreviated keys are possible in yaml, wrong keys are indeed ignored by github already. You seem to ignore messages like `Warning: Unexpected input(s) 'remove-output', 'mingw64'` and so I don't know when you saw what, and what result it gave. The plugin seems enabled from what I looked at, but I cannot really work with fluent things like that.

**Comment by kayhayen** [comment]
I added the whole C backend selection group, as it also controls a few interesting features like CF protection, but I cannot really see the issue, Nuitka Options seems to report the options as used. If you enabled the report, we could see if the plugin is actually enabled, but alas, I feel that option group is also missing or what.

**Comment by Jaxx7594** [comment]
> I am not sure what you think that screenshot is going to help with. I indeed removed a difference between those two, I wasn't sure what actually happens, but I corrected the crash I saw, under the theory that abbreviated keys are possible in yaml, wrong keys are indeed ignored by github already. You seem to ignore messages like `Warning: Unexpected input(s) 'remove-output', 'mingw64'` and so I don't know when you saw what, and what result it gave. The plugin seems enabled from what I looked at, but I cannot really work with fluent things like that.

I just thought it might be useful to provide visual proof that the tcl files that would normally be there aren't present. Also, the Nuitka console outputs that normally show that the tk-inter plugin is enabled isn't printed (something along the lines of "tk-inter plugin: Added tcl files" that appears at the very end of the console outputs, can't check what exactly it says right now). And from my testing it seems github passes unexpected inputs through anyways. For example, when building with the remove-output input, the .build folder isn't created, even though its an unexpected input. I don't really know what to make of the other things you said in your message, as I very rarely use github, and have never used github actions nor yml files, so I am very unknowledgeable about this.

**Comment by kayhayen** [comment]
Can you add the new key `report` and provide it as an artifact as every compilation ought to do, maybe with 'report-diffable' to true, as it will make it easy to diff them. From the report we can tell what plugins got activated or not.

Interesting, that would still work. I must admit, I am not nearly an expect, but I got the idea that unknown keys are just silently dropped. Also the .build folder is always created, it will just be removed. On GitHub actions removing it makes not a lot of sense, since that thing is ephemeral anyway there.

**Comment by kayhayen** [comment]
I checked now, and you are indeed correct, unknown options are passed. It also explains why your had the crash you had, where I was initially convinced it would be impossible. That however means, that abbreviations like that need to be detected by Nuitka, as e.g. `--min` will work for as long as it's unique, but with Nuitka not recognizing them, where they have special meaning, then of course, hell breaks loose. You report already gave rise to the suspicion, but your action log confirms it for entirely unknown options. However only these spellings will have an effect on the command line parsing:

```
    early_names = (
        "main",
        "script-name",
        "enable-plugin",
        "enable-plugins",
        "disable-plugin",
        "disable-plugins",
    )
```

I added 2 of those now, rest of Nuitka does e.g. these:

```
        if arg.startswith(
            ("--enable-plugin=", "--enable-plugins=", "--plugin-enable=")
        ):
```

and only these. Adding command line argument support for plugins based on present options is always going to be a hack.

We now have 2 options, the plugin still doesn't get activated for you. Or it gets activated, but doesn't work on GitHub Actions very well.
